From: Philipp Zabel <p.zabel@pengutronix.de>
Date: Wed, 3 Apr 2019 17:08:53 +0200
Subject: [PATCH] media: coda: let decoder command return busy error during
 draining

While in draining state, let VIDIOC_DECODER_CMD return -EBUSY as
required by the V4L2 stateful decoder API specification.

Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
---
 drivers/media/platform/chips-media/coda-common.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/media/platform/chips-media/coda-common.c b/drivers/media/platform/chips-media/coda-common.c
index 0afd737eb935..e0330bca8508 100644
--- a/drivers/media/platform/chips-media/coda-common.c
+++ b/drivers/media/platform/chips-media/coda-common.c
@@ -1406,6 +1406,12 @@ static int coda_decoder_cmd(struct file *file, void *fh,
 	if (ret < 0)
 		return ret;
 
+	if (ctx->state == CODA_STATE_DRAIN) {
+		coda_dbg(1, ctx, "dec_cmd: invalid state: %s\n",
+			 coda_state_name(ctx->state));
+		return -EBUSY;
+	}
+
 	switch (dc->cmd) {
 	case V4L2_DEC_CMD_START:
 		mutex_lock(&dev->coda_mutex);
